language/move-binary-format/src/check_bounds.rs

```diff
@@ -19,100 +20,132 @@ use move_core_types::vm_status::StatusCode;
 use std::u8;
 
 pub struct BoundsChecker<'a> {
-    module: &'a CompiledModuleMut,
+    view: &'a BinaryIndexedViewMut<'a>,
     current_function: Option<FunctionDefinitionIndex>,
 }
 
 impl<'a> BoundsChecker<'a> {
-    pub fn verify(module: &'a CompiledModuleMut) -> PartialVMResult<()> {
+    pub fn verify_script(view: &'a BinaryIndexedViewMut) -> PartialVMResult<()> {
```

Comment:
Sounds good, I'll try to do this. I went this way because, before speaking with @vgao1996 earlier today, I thought the `CompiledScriptMut` vs `CompiledScript` distinction was a useful one that we wanted to keep in this codebase. Since the only difference between `CompiledScriptMut` and `CompiledScript` is that the latter has been bounds-checked, I thought there simply wasn't a way to get a `CompiledScript` from a `CompiledScriptMut` without calling `CompiledScriptMut::freeze` and performing bounds checking on the `CompiledScriptMut`.

But now, knowing the distinction between mut vs non-mut is meant to be eliminated, I may try to eliminate the `CompiledScriptMut::freeze` method altogether, since it seems like the entire point is to run the bounds checker and return a non-mut object.